File: 4673-selfNumber.py
def selfNum(num) :
    sum = int(num)
    for i in range(0,4):
        if num == 0:
            break
        else:
            sum += int(num) % 10
            num /= int(10)
    return sum

arr = []
for i in range(1, 10001):
    key = selfNum(i)
    #print(key)
    arr.append(key)
    if i not in arr:
        print(i)
   
# arr = []
# for i in range(1, 10001):
#     #print(key)
#     sum = i
#     num = i
#     for j in range(0,4):
#         if num == 0:
#             break
#         else:
#             sum += int(num) % 10
#             num /= int(10)
#     arr.append(sum)
    
#     if i not in arr:
#         print(i)

   

File: 4673-selfNumber.cpp
#include <stdio.h>
#define MAX 10001

int selfNum(int num) {
    int sum = num;
    for(int i=0; i<4; i++) { 
        if(num == 0)
            break;
        else {
            sum += num % 10;
            num /= 10;
        }
    }
    return sum;
}

int main(void) {
    int arr[MAX] = {0,};
    for(int i=1; i<MAX; i++) {
        int key = selfNum(i);
        if(key < MAX) {
            arr[key] = 1;
        }
        // printf("%d\n", key);
        if(arr[i] == 0)
            printf("%d\n", i);
    }
    return 0;
}

   